blog/2019-03-14-debezium-meets-quarkus.adoc

```diff
@@ -0,0 +1,96 @@
+= Debezium meets Quarkus
+jpechane
+:awestruct-tags: [ quarkus, examples, microservices, apache-kafka ]
+:awestruct-layout: blog-post
+
+[role="teaser"]
+--
+A new community project for developing cloud-native applications https://quarkus.io/[Quarkus] has been launched.
+We are going to demonstrate how a Quarkus-based application could be used to consume event messages created with link:2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern[outbox pattern].
+--
+
+== About Quarkus
+https://quarkus.io/[Quarkus] is a Java stack designed for development of cloud-native applications based on Java platform.
+It provides a subset of best-of-breed technologies and libraries that are neatly integrated together.
+It supports a development of both imperative and reactive applications with support for mixing of both approaches together.
+It is designed for significantly reduced memory consumption and improved startup time.
+
+Last but not least it supports both OpenJDK HotSpot and GraalVM virtual machines.
+With GraalVM it is possible to compile the application into a native binary and thus reduce the resource consumption and startup time even more.
+
+== Service implementation
+
+We have extended the https://github.com/debezium/debezium-examples/tree/master/outbox[example application] introduced in the link:2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern[recent blog post] with a new service named `shipment-service-quarkus`.
+It provides the same functionality as the `shipment-service` does with few implementation differences
+
+ * Quarkus right now supports only MariaDB database, not MySQL, so we have included an instance on MariaDB database to which the service is writing
+ * JSON-P API is not supported yet, the JSON parser of choice is Jackson. The code has been modified to use Jackson API instead of JSON-P API.
+ * Quarkus supports https://github.com/eclipse/microprofile-reactive-messaging[Reactive Messaging for MicroProfile] via http://smallrye.io/[SmallRye] implementation. We no longer need to use Kafka client as the message listener is implemented declaratively.
+
+== Build process
+
+The build process is almost the same as was original.
+There are two changes though
```

Comment:
Also let's show a snippet from the _pom.xml_?

Reply:
What the snippet should show? Quarkus plug-in configuration?

Reply:
Yes, the Quarkus plug-in config. So people get a feel of what it takes.